[IMP] accounting: batch payments and SEPA overhaul (+ additional CH loca info) (#17751)
odoo/documentation: content/applications/finance/accounting/payments/batch.rst

Batch payments
==============

Batch payments act as an organizational tool that groups multiple individual customer or vendor
payments into a single file. Rather than managing transactions one by one, you can consolidate them
to generate a detailed deposit slip or a single electronic payment file. This significantly

> simplifies bank :doc:`reconciliation <../bank/reconciliation>`: when a bulk sum appears on your bank

Reviewer: ```suggestion
simplifies :doc:`bank reconciliation <../bank/reconciliation>`: when a bulk sum appears on your bank
```

> simplifies bank :doc:`reconciliation <../bank/reconciliation>`: when a bulk sum appears on your bank

Reviewer 2: ```suggestion
simplifies :doc:`bank reconciliation <../bank/reconciliation>`: when a bulk sum appears on your bank
```

statement, Odoo uses the batch reference to automatically match and reconcile all underlying
payments in a single step.

> For customer payments, the feature supports both physical assets, such :doc:`checks <checks>` and

Reviewer: ```suggestion
For customer payments, the feature supports both physical assets, such as :doc:`checks <checks>` and
```

> For customer payments, the feature supports both physical assets, such :doc:`checks <checks>` and

Reviewer 2: ```suggestion
For customer payments, the feature supports both physical payments, such :doc:`checks <checks>` and
```
Let's adjust this so it's not confused with asset management :)

cash, and electronic transfers. You can group daily receipts into one batch to print a bank deposit

> slip, ensuring your internal records mirror the lump-sum deposit on your bank statement. On the

Reviewer 2: ```suggestion
cash, and electronic transfers. Group incoming cash or check payments into batches to print bank deposit
slips to ensure your internal records mirror the lump-sum deposit on your bank statement. On the
```
Actually, I think it's fine to leave this point about bank deposit slips, but if you feel differently, feel free to change it. But we can eliminate some instances of "you" and make "daily receipts" more clear.

vendor side, batch payments streamline bulk electronic transfers. Instead of processing supplier

> invoices manually, you can select multiple bills and generate a single outgoing payment file to

Reviewer 2: ```suggestion
invoices manually, select multiple bills and generate a single outgoing payment file to
```

upload directly to your banking portal. Because these file formats are region-specific (e.g.,
:doc:`SEPA <sepa_payments>` in Europe, :ref:`NACHA <l10n_us/nacha>` in the U.S., or the global

> :ref:`ISO 20022 <accounting/sepa_payments/iso20022>` standard), you should consult your country's

Reviewer 2: ```suggestion
:ref:`ISO 20022 <accounting/sepa_payments/iso20022>` standard), consult your country's
```

:doc:`fiscal localization <../../fiscal_localizations>` documentation to verify supported formats.

.. seealso::

>    - :doc:`../payments`

Reviewer: ```suggestion
   :doc:`../payments`
```
Bullet point isn't necessary since there is only one element :)

.. _accounting/batch/configuration:

Configuration
=============

> To enable batch payments, open the **Accounting** app, go to :menuselection:`Configuration -->

Reviewer: Batch payments are enabled by default, on Odoo Online at least, which is our reference.
What do you think about phrasing it this way to incorporate that fact (because it's still interesting to mention): _Ensure batch payments are enabled by going to (...)_ ?

Settings`, scroll down to the :guilabel:`Customer Payments` section, and enable :guilabel:`Batch
Payments`.

.. note::
   This enables both customer *and* vendor batch payments.

.. tip::
   According to your needs, check that the following apps are installed in the **Apps** app:

   - Batch Payment

>    - Account Batch Payment Reconciliation

Reviewer: These two modules are installed by default with the batch payment option enabled, so I wouldn't mention them. For the other modules, I guess it depends on your needs/localization, but the tip isn't really useful as it is. Maybe with some links to the relevant/related docs?

   - SEPA Credit Transfer
   - SEPA Direct Debit
   - SEPA Payments for Payroll
   - Payment Provider: Sepa Direct Debit
   - NACHA Payments

.. _accounting/batch/creation:

Batch creation
==============

> To create customer or vendor batch payments, reproduce the following steps:

Reviewer 2: ```suggestion
To create customer or vendor batch payments and add existing payments to them, follow these steps:
```
Let's summarize each of the two options, also "reproduce the following steps" sounds overly formal

#. Make sure all payments to be included in the batch have been :ref:`registered
   <accounting/payments/from-invoice-bill>`.
#. Open the **Accounting** app and go to :menuselection:`Customers --> Batch Payments` or
   :menuselection:`Vendors --> Batch payments` according to the nature of the payment.

> #. Click :guilabel:`New`. From here, some configuration is required:

Reviewer: ```suggestion
   :menuselection:`Vendors --> Batch payments` according to the nature of the payment and click :guilabel:`New`.
   #. Next, configure the batch payment:
```
I would include the _click New_ with the first bullet point, it feels like it's a unique action.
I tried to be a tiny bit more precise for this sentence _From here, some configuration is required:_

> #. Click :guilabel:`New`. From here, some configuration is required:

Reviewer 2: ```suggestion
#. Click :guilabel:`New`. Complete the following fields:
```

   - :guilabel:`Batch Type`: select whether the money is being transferred to your account

>      (:guilabel:`Inbound`) or to somebody else's account (:guilabel:`Outbound`).

Reviewer 2: ```suggestion
     (:guilabel:`Inbound`) or to someone else's account (:guilabel:`Outbound`).
```
"somebody" is more common when speaking and less formal but "someone" is better when writing

   - :guilabel:`Bank`: the bank journal to use for this batch.
   - :guilabel:`Payment Method`: the payment method used for the invoices' payments. Only payments
     matching the payment method selected will appear.
   - :guilabel:`Date`: the batch's creation date

>    - :guilabel:`Reference`: the reference of the batch. Leave it empty to generate it automatically.
>    - :guilabel:`Add a line`: click it to select the payments to include in the batch.
> #. Once all desired payments are included, click :guilabel:`Validate` to finalize the batch.

Reviewer: ```suggestion
   - :guilabel:`Add a line`: click it to select the payments to include in the batch.

#. Once all desired payments are included, click :guilabel:`Validate` to finalize the batch.
```

> Alternatively, you can:

Reviewer 2: ```suggestion
Alternatively, select the relevant payments and create a batch payment directly from them:
```
Let's summarize each of the two options

#. Make sure all payments to be included in the batch have been :ref:`registered
   <accounting/payments/from-invoice-bill>`.
#. Open the **Accounting** app and go to :menuselection:`Customers --> Payments` or
   :menuselection:`Vendors --> Payments` and select all payments to include in the batch.
#. Click :guilabel:`Create Batch`, *or* :icon:`fa-cog` :guilabel:`(Actions)`, and select
   :guilabel:`Create batch payment`.
#. In the view form, review the selected payments. If any individual payments are missing, click
   :guilabel:`Add a line`, then select the missing payments to include in the batch.
#. Once all desired payments are included, click :guilabel:`Validate` to finalize the batch.

To view existing batch payments, go to :menuselection:`Customers --> Batch Payments` or
:menuselection:`Vendors --> Batch Payments`.

.. note::
   - All payments in a batch *must* use the same payment method.
   - Once validated, no additional payments can be added to a batch. You can delete the batch if
     necessary by clicking :icon:`fa-cog` :guilabel:`(Actions)` and then :icon:`fa-trash-o`
     :guilabel:`(Delete)`.

.. tip::

>    - Click :guilabel:`Print` to download a list of the included payments.

Reviewer: ```suggestion
   - Click :icon:`fa-print` :guilabel:`Print` to download a list of the included payments.
```
Let's add that icon while we are there :)

   - To filter payments by payment method, click on the :guilabel:`Payment Method` column header
     during the batch payment creation step.

> 

Reviewer: ```suggestion
```

.. _accounting/batch/bank_reconciliation:

Bank reconciliation
===================

> Once the bank transactions :doc:`have been created <../bank/transactions>` in your database, you can

Reviewer 2: ```suggestion
Once the :doc:`bank transactions <../bank/transactions>` have been created in your database,
```

Batch payments
==============

Batch payments act as an organizational tool that groups multiple individual customer or vendor
payments into a single file. Rather than managing transactions one by one, you can consolidate them
to generate a detailed deposit slip or a single electronic payment file. This significantly

> simplifies :doc:`bank reconciliation <../bank/reconciliation>`: when a bulk sum appears on your bank

Reviewer 2: Note that in 19, I don't think you're able to reconcile against batch payments anymore, so some of this intro and the bank rec section will need to change

.. note::
   If a specific payment could not be processed by the bank or is missing, remove the line
   corresponding to that payment by using the :icon:`fa-trash-o` (:guilabel:`Delete`) button before
   validating the reconciliation of the batch payment.

.. _accounting/batch/bank-accounts:

Bank accounts
=============

Odoo distinguishes between :guilabel:`Banks` (the financial institution issuing the account) and
:guilabel:`Bank Accounts` (the specific account number and routing details).

You can store banking details for both your own company and your :doc:`contacts
<../../../essentials/contacts>`. Odoo automatically uses this information when needed, such as when
generating batch payment files.

To protect your business against :ref:`phishing scams <accounting/batch/phishing>`, Odoo configures
all new bank accounts as "untrusted" by default, preventing outgoing wire transfers until you
:ref:`manually verify and trust <accounting/batch/link-account-contact>` the account.

.. seealso::
   :doc:`sepa_payments`
   :ref:`NACHA <l10n_us/nacha>`

.. _accounting/batch/bank-accounts-configuration:

Configuration
-------------

.. note::

>    The **Contacts** app can be downloaded from the **Apps** dashboard.

Reviewer 2: ```suggestion
   The **Contacts** app can be :ref:`installed <general/install>` from the **Apps** dashboard.
```

- :guilabel:`Bank Address`: Enter the :guilabel:`Country`. The :guilabel:`Street`, :guilabel:`City`,
  :guilabel:`State`, and :guilabel:`Zip` fields are optional.
- :guilabel:`Phone` and :guilabel:`Email`: Optional contact details.

.. _accounting/batch/create-bank-account:

Create a bank account
~~~~~~~~~~~~~~~~~~~~~

.. note::
   **Prerequisite:** A bank account must be linked to a specific bank record. Ensure you have
   created the required bank using the steps above before setting up a bank account.

Go to :menuselection:`Configuration --> Bank Accounts` and click :guilabel:`New`. Enter the
following details:

- :guilabel:`Account Number`: The unique identification number for the bank account.
- :guilabel:`Bank`: Select the associated bank from the drop-down menu.
- :guilabel:`ABA/Routing`: The American Bankers Association routing number (if applicable).
- :guilabel:`Account Holder Name`: The full name of the account holder.
- :guilabel:`Account Holder`: Select the holder's :doc:`contact <../../../essentials/contacts>` from
  the drop-down menu. If the contact does not exist yet, you can link it later by following the next
  steps.

> 

Reviewer 2: Can you add the "Trusted" option here too? It can be set even if there isn't a contact that it's linked to

.. _accounting/batch/link-account-contact:

Link a bank account to a contact
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

If you need to link an existing bank account to a contact:

1. Open an app with contact access (such as **Accounting** or **Contacts**).
2. Select a contact and open the :guilabel:`Accounting` tab.
3. In the :guilabel:`Bank Accounts` section, click :guilabel:`Add a line` and provide the:

   - :guilabel:`Account Number`,
   - :guilabel:`Bank`,
   - :guilabel:`Account Holder Name`,
   - :guilabel:`Account Holder`,
   - :guilabel:`Company` and :guilabel:`Currency` (if applicable).

> 4. Toggle :guilabel:`Send Money` to mark the account as :guilabel:`Trusted`.

Reviewer 2: ```suggestion
4. Toggle :guilabel:`Send Money` to mark the account as :guilabel:`Trusted`. Marking the account as :guilabel:`Trusted` will enable you to send money to this account, so only do this when you're absolutely sure the information is correctly entered and was not provided by a :ref:`phishing attack <accounting/batch/phishing>`.
```
An explanation of what exactly this field does/what it's for would be helpful, as well as a connection to the phishing attacks section so it's clear how the two are linked. Feel free to edit my suggestion, and I think something similar should at least be put in the section above where I suggested adding the Trusted option, and if you want to include it here also or just there is up to you.
